Tests/Packet++Test/Tests/S7CommTests.cpp

```diff
@@ -0,0 +1,72 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "EndianPortable.h"
+#include "Packet.h"
+#include "S7CommLayer.h"
+#include "SystemUtils.h"
+
+PTF_TEST_CASE(S7CommLayerTest)
+{
+	timeval time;
+	gettimeofday(&time, nullptr);
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/S7comm.dat");
+
+	pcpp::Packet S7CommLayerTest(&rawPacket1);
+	PTF_ASSERT_TRUE(S7CommLayerTest.isPacketOfType(pcpp::S7COMM));
+	auto *S7CommLayer = S7CommLayerTest.getLayerOfType<pcpp::S7CommLayer>();
+	PTF_ASSERT_NOT_NULL(S7CommLayer);
+
+	PTF_ASSERT_EQUAL(S7CommLayer->getProtocolId(), 0x32);
+	PTF_ASSERT_EQUAL(S7CommLayer->getMsgType(), 0x07);
+	PTF_ASSERT_EQUAL(S7CommLayer->getPduRef(), 0xfd0b);
+	PTF_ASSERT_EQUAL(S7CommLayer->getParamLength(), 12);
+	PTF_ASSERT_EQUAL(S7CommLayer->getDataLength(), 212);
+	PTF_ASSERT_EQUAL(S7CommLayer->getHeaderLen(), 0xea);
+	PTF_ASSERT_EQUAL(S7CommLayer->toString(), "S7Comm Layer, Job Request");
+
+	PTF_ASSERT_EQUAL(S7CommLayer->getParameter()->getDataLength(), 12);
+	uint8_t expectedParameterData[] = {0x00, 0x01, 0x12, 0x08, 0x12, 0x84, 0x01, 0x01, 0x00, 0x00, 0x00, 0x00};
+	PTF_ASSERT_BUF_COMPARE(S7CommLayer->getParameter()->getData(), expectedParameterData, 12);
+
+	pcpp::S7CommLayer newS7commPacket(0x09, 0xfd0c, 13, 213);
```

Comment:
Is it possible to separate this to a different test `S7CommLayerCreationTest`?

We can rename this test to `S7CommLayerParsingTest`